### PR TITLE
VB-608 Refactor main scheduler service

### PR DESCRIPTION
**Pull request by philip-milne**
## What does this pull request do?
Refactor of the main scheduler service into separate services
Removed redundant table and scripts, impl, tests and debug containing personal information
Minor change to swagger docs for consistency

## What is the intent behind these changes?
To make the services more cohesive and easier to maintain.
Separate concerns & improve code quality
Remove dead code and invalid tests

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResource.kt [hunk 1]
```diff
@@ -173,11 +173,10 @@ class VisitResource(
       ),
     ]
   )
-  fun getVisitById(
+  fun getVisitByReference(
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResource.kt:176]: woops must have missed some of these

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResource.kt [hunk 2]
```diff
@@ -145,7 +145,7 @@ class VisitResource(
   @GetMapping("/{reference}")
   @Operation(
     summary = "Get visit",
-    description = "Retrieve visit by visit id",
+    description = "Retrieve visit by visit reference",
     responses = [
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResource.kt:149]: woops I must have missed some of these

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitSessionTemplateResource.kt [hunk 3]
```diff
@@ -19,14 +19,14 @@ import org.springframework.web.bind.annotation.RestController
...
 import uk.gov.justice.digital.hmpps.visitscheduler.data.CreateSessionTemplateRequest
 import uk.gov.justice.digital.hmpps.visitscheduler.data.SessionTemplateDto
-import uk.gov.justice.digital.hmpps.visitscheduler.service.VisitSchedulerService
+import uk.gov.justice.digital.hmpps.visitscheduler.service.SessionTemplateService
 import javax.validation.Valid
 
 @RestController
 @Validated
-@RequestMapping(name = "Visit Resource", path = ["/visit-session-templates"], produces = [MediaType.APPLICATION_JSON_VALUE])
+@RequestMapping(name = "Session Resource", path = ["/visit-session-templates"], produces = [MediaType.APPLICATION_JSON_VALUE])
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitSessionTemplateResource.kt:27]: may be Session Template Resource

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt [hunk 4]
```diff
@@ -0,0 +1,175 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.service
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt:1]: I assume this is just a rename

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionTemplateService.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.service
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionTemplateService.kt:1]: I assume this is just a rename

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt (hunk 4 above)
Comment (philip-milne) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt:1]: split the visit, session, session-template & support.  session and session-template are really the same thing but split it based on the endpoints.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitSessionTemplateResource.kt (hunk 3 above)
Comment (philip-milne) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitSessionTemplateResource.kt:27]: The session and Session Template resource are really the same, do you want me to update for consistency with the service names?

**Review by AledWEvans — APPROVED** [review]

**Pull request merged**